### ISSUE TITLE
The OMD of PowerBI-Visuals-API

### ISSUE THREAD

**Issue by luojiandan** [JorgeCandeias/Power-BI-Custom-Visual-Fundamentals#2]
Hi,Jorge,I watch this vedio,and I have a question,I can't find the "Object Model Diagram" for PowerBI-Visuals-API,like this:(link: https://desktop.arcgis.com/en/arcobjects/latest/java/diagrams/engine/CartoObjectModel.pdf) desktop.arcgis.com/en/arcobjects/…
can you help me?

**Comment by JorgeCandeias** [comment]
I'm not aware whether there is one. The closest thing to API documentation I'm aware of is [this](https://microsoft.github.io/PowerBI-visuals/api/references/ivisualhost/) and [this](https://microsoft.github.io/PowerBI-visuals/docs/concepts/capabilities/). However, I haven't touched custom visuals in ages so I'm not sure what is the current state of things.

**Comment by luojiandan** [comment]
> I'm not aware whether there is one. The closest thing to API documentation I'm aware of is [this](https://microsoft.github.io/PowerBI-visuals/api/references/ivisualhost/) and [this](https://microsoft.github.io/PowerBI-visuals/docs/concepts/capabilities/). However, I haven't touched custom visuals in ages so I'm not sure what is the current state of things.
yes,my question came from [this](https://microsoft.github.io/PowerBI-visuals/docs/how-to-guide/migrating-to-powerbi-visuals-tools-3-0/):
these are many interface not see before,but this document not mention too,I can't image the whole structure of api,it's hard to understand which interface can be used when code.
`import PrimitiveValue = powerbi.PrimitiveValue; 
import VisualUpdateOptions = powerbi.extensibility.visual.VisualUpdateOptions; 
import VisualConstructorOptions = powerbi.extensibility.visual.VisualConstructorOptions; 
import IVisualHost = powerbi.extensibility.visual.IVisualHost; 
import IColorPalette = powerbi.extensibility.IColorPalette; 
import IVisual = powerbi.extensibility.visual.IVisual; 
import VisualObjectInstance = powerbi.VisualObjectInstance; 
import VisualObjectInstanceEnumeration = powerbi.VisualObjectInstanceEnumeration; 
import EnumerateVisualObjectInstancesOptions = powerbi.EnumerateVisualObjectInstancesOptions; 
import Fill = powerbi.Fill; 
import VisualTooltipDataItem = powerbi.extensibility.VisualTooltipDataItem; 
import ISelectionManager = powerbi.extensibility.ISelectionManager; `
